### PR TITLE
First pass at surviving certain TLS connection issues. See #696.

### PR DESCRIPTION
**Pull request by meitar**
This is probably a kludge because I am not super familiar with TLS
socket programming or SET generally, but it achieves the result, which
is not to shutdown the (HTTPS) socket server when a TLS client responds
to a TLS Server Hello message with a fatal Alert message.

One example of a client that does this is recent Firefox. What this
means is that if you run SET with `WEBATTACK_SSL=ON` and
`SELF_SIGNED_CERT=ON`, your victim can turn off your HTTPS server simply
by navigating to your attack page.

This is caused by the underlying OpenSSL library raising an error that
`pyopenssl`, in turn, `raise`s through the socket server libraries.
Ultimately, it bubbles up to the `harvester` module through its
`shutdown_request` method, called by the underlying socket server's
`_handle_request_noblock` method. See the backtrace printed in the
comments of Pull Request #696 for a complete example.

The bottom line is that this unhandled exception ultimately causes the
HTTPS server to die before it gets a chance to be useful. Since I assume
that SET doesn't particularly care what certificate validation alerts the
client is sending us, this patch addresses the issue by ignoring every
raised exception from the underlying libraries except for a
`KeyboardInterrupt` so that the SET user can cause a server shutdown
themselves, with the expected `C-c` interrupt signal.

There is probably a more graceful way to handle this, though? Also, note
that this only fixes the HTTPS issues for Python 3. Python 2 exhibits a
different error altogether.

### CODE REVIEW

**Comment by meitar** [comment]
Here is some additional detail about the issue. The file [`decrypted-fatal-from-firefox.pcapng.zip`](https://github.com/trustedsec/social-engineer-toolkit/files/4314970/decrypted-fatal-from-firefox.pcapng.zip) is a packet capture of the TLS conversation between more or less default Firefox 74.0 installation and SET before this patch was applied. The important detail is in packet number 3, which shows a "`Alert (Level: Fatal, Description: Unknown CA)`", and this maps to the OpenSSL error raised `pyopenssl`, which is "`OpenSSL.SSL.Error: [('SSL routines', 'ssl3_read_bytes', 'tlsv1 alert unknown ca')]`".
